Failing to compile js

Compiling starling master produces build errors in js:
```
   Compiling js v0.1.4 (https://github.com/fitzgen/mozjs?branch=smup-smup-smup#d1e34d61)
error[E0592]: duplicate definitions with name `succeed`
    --> /home/ajeffrey/.cargo/git/checkouts/mozjs-f94f19fa8f916dc5/d1e34d6/mozjs/js/rust/src/lib.rs:50:5
     |
50   | /     pub fn succeed(&mut self) -> bool {
51   | |         self.code_ = JS::ObjectOpResult_SpecialCodes::OkCode as usize;
52   | |         true
53   | |     }
     | |_____^ duplicate definitions for `succeed`
     | 
    ::: /home/ajeffrey/github/starling/target/debug/build/js-e6b8781f779e8d2f/out/jsapi.rs
     |
4450 | /             pub unsafe fn succeed(&mut self) -> bool {
4451 | |                 ObjectOpResult_succeed(self)
4452 | |             }
     | |_____________- other definition for `succeed`

error[E0592]: duplicate definitions with name `new`
    --> /home/ajeffrey/.cargo/git/checkouts/mozjs-f94f19fa8f916dc5/d1e34d6/mozjs/js/rust/src/rust.rs:531:5
     |
531  | /     pub fn new() -> JS::HandleValueArray {
532  | |         JS::HandleValueArray {
533  | |             length_: 0,
534  | |             elements_: ptr::null(),
535  | |         }
536  | |     }
     | |_____^ duplicate definitions for `new`
     | 
    ::: /home/ajeffrey/github/starling/target/debug/build/js-e6b8781f779e8d2f/out/jsapi.rs
     |
5021 | /             pub unsafe fn new(value: *const root::JS::RootedValue) -> Self {
5022 | |                 let mut __bindgen_tmp = ::std::mem::uninitialized();
5023 | |                 HandleValueArray_HandleValueArray(&mut __bindgen_tmp, value);
5024 | |                 __bindgen_tmp
5025 | |             }
     | |_____________- other definition for `new`

error[E0592]: duplicate definitions with name `rval`
     --> /home/ajeffrey/.cargo/git/checkouts/mozjs-f94f19fa8f916dc5/d1e34d6/mozjs/js/rust/src/rust.rs:741:5
      |
741   | /     pub fn rval(&self) -> JS::MutableHandleValue {
742   | |         self._base
743   | |     }
      | |_____^ duplicate definitions for `rval`
      | 
     ::: /home/ajeffrey/github/starling/target/debug/build/js-e6b8781f779e8d2f/out/jsapi.rs
      |
15253 | /         pub unsafe fn rval(&mut self) -> root::JS::MutableHandleValue {
15254 | |             JSJitGetterCallArgs_rval(self)
15255 | |         }
      | |_________- other definition for `rval`

error[E0592]: duplicate definitions with name `get`
     --> /home/ajeffrey/.cargo/git/checkouts/mozjs-f94f19fa8f916dc5/d1e34d6/mozjs/js/rust/src/rust.rs:634:5
      |
634   | /     pub fn get(&self, i: u32) -> JS::HandleValue {
635   | |         unsafe {
636   | |             if i < self._base.argc_ {
637   | |                 JS::HandleValue::from_marked_location(self._base.argv_.offset(i as isize))
...     |
641   | |         }
642   | |     }
      | |_____^ duplicate definitions for `get`
      | 
     ::: /home/ajeffrey/github/starling/target/debug/build/js-e6b8781f779e8d2f/out/jsapi.rs
      |
15390 | /         pub unsafe fn get(&self, i: ::std::os::raw::c_uint)
15391 | |          -> root::JS::HandleValue {
15392 | |             JSJitMethodCallArgs_get(self, i)
15393 | |         }
      | |_________- other definition for `get`

error[E0592]: duplicate definitions with name `rval`
     --> /home/ajeffrey/.cargo/git/checkouts/mozjs-f94f19fa8f916dc5/d1e34d6/mozjs/js/rust/src/rust.rs:661:5
      |
661   | /     pub fn rval(&self) -> JS::MutableHandleValue {
662   | |         unsafe {
663   | |             JS::MutableHandleValue::from_marked_location(self._base.argv_.offset(-2))
664   | |         }
665   | |     }
      | |_____^ duplicate definitions for `rval`
      | 
     ::: /home/ajeffrey/github/starling/target/debug/build/js-e6b8781f779e8d2f/out/jsapi.rs
      |
15374 | /         pub unsafe fn rval(&self) -> root::JS::MutableHandleValue {
15375 | |             JSJitMethodCallArgs_rval(self)
15376 | |         }
      | |_________- other definition for `rval`

error: aborting due to 5 previous errors

error: Could not compile `js`.
```

Removed the duplicates, now getting a different error:
```
   Compiling js v0.1.4 (file:///home/ajeffrey/github/asajeffrey/mozjs/mozjs/js/rust)
error[E0394]: cannot refer to other statics by value, use the address-of operator or a constant instead
    --> /home/ajeffrey/github/asajeffrey/mozjs/mozjs/js/rust/src/rust.rs:1015:13
     |
1015 |     flags: (JSCLASS_IS_GLOBAL | ((JSCLASS_GLOBAL_SLOT_COUNT & JSCLASS_RESERVED_SLOTS_MASK) << JSCLASS_RESERVED_SLOTS_SHIFT)) as u32,
     |             ^^^^^^^^^^^^^^^^^ referring to another static by value
     |
     = note: use the address-of operator or a constant instead

error[E0394]: cannot refer to other statics by value, use the address-of operator or a constant instead
    --> /home/ajeffrey/github/asajeffrey/mozjs/mozjs/js/rust/src/rust.rs:1015:35
     |
1015 |     flags: (JSCLASS_IS_GLOBAL | ((JSCLASS_GLOBAL_SLOT_COUNT & JSCLASS_RESERVED_SLOTS_MASK) << JSCLASS_RESERVED_SLOTS_SHIFT)) as u32,
     |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^ referring to another static by value
     |
     = note: use the address-of operator or a constant instead

error[E0394]: cannot refer to other statics by value, use the address-of operator or a constant instead
    --> /home/ajeffrey/github/asajeffrey/mozjs/mozjs/js/rust/src/rust.rs:1015:63
     |
1015 |     flags: (JSCLASS_IS_GLOBAL | ((JSCLASS_GLOBAL_SLOT_COUNT & JSCLASS_RESERVED_SLOTS_MASK) << JSCLASS_RESERVED_SLOTS_SHIFT)) as u32,
     |                                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^ referring to another static by value
     |
     = note: use the address-of operator or a constant instead

error: use of extern static requires unsafe function or block (error E0133)
    --> /home/ajeffrey/github/asajeffrey/mozjs/mozjs/js/rust/src/rust.rs:1015:13
     |
1015 |     flags: (JSCLASS_IS_GLOBAL | ((JSCLASS_GLOBAL_SLOT_COUNT & JSCLASS_RESERVED_SLOTS_MASK) << JSCLASS_RESERVED_SLOTS_SHIFT)) as u32,
     |             ^^^^^^^^^^^^^^^^^
     |
     = note: #[deny(safe_extern_statics)] on by default
     = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
     = note: for more information, see issue #36247 <https://github.com/rust-lang/rust/issues/36247>

error: use of extern static requires unsafe function or block (error E0133)
    --> /home/ajeffrey/github/asajeffrey/mozjs/mozjs/js/rust/src/rust.rs:1015:35
     |
1015 |     flags: (JSCLASS_IS_GLOBAL | ((JSCLASS_GLOBAL_SLOT_COUNT & JSCLASS_RESERVED_SLOTS_MASK) << JSCLASS_RESERVED_SLOTS_SHIFT)) as u32,
     |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^
     |
     = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
     = note: for more information, see issue #36247 <https://github.com/rust-lang/rust/issues/36247>

error: use of extern static requires unsafe function or block (error E0133)
    --> /home/ajeffrey/github/asajeffrey/mozjs/mozjs/js/rust/src/rust.rs:1015:63
     |
1015 |     flags: (JSCLASS_IS_GLOBAL | ((JSCLASS_GLOBAL_SLOT_COUNT & JSCLASS_RESERVED_SLOTS_MASK) << JSCLASS_RESERVED_SLOTS_SHIFT)) as u32,
     |                                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
     |
     = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
     = note: for more information, see issue #36247 <https://github.com/rust-lang/rust/issues/36247>

error: use of extern static requires unsafe function or block (error E0133)
    --> /home/ajeffrey/github/asajeffrey/mozjs/mozjs/js/rust/src/rust.rs:1039:19
     |
1039 |     class.flags & JSCLASS_IS_DOMJSCLASS != 0
     |                   ^^^^^^^^^^^^^^^^^^^^^
     |
     = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
     = note: for more information, see issue #36247 <https://github.com/rust-lang/rust/issues/36247>

error: aborting due to 7 previous errors

error: Could not compile `js`.
```
The generated `jsapi.rs` file includes:
```rust
    extern "C" {
        #[link_name = "_ZL17JSCLASS_IS_GLOBAL"]
        pub static JSCLASS_IS_GLOBAL: u32;
    }
```
so yes, it's a `static` rather than a `const`.

--- a/README.md
+++ b/README.md
@@ -3,3 +3,17 @@
 [![Build Status](https://travis-ci.org/starlingjs/starling.png?branch=master)](https://travis-ci.org/starlingjs/starling)
 
 The Starling JavaScript runtime.
+
+## Building
+
+To build starling, you need `autoconf` 2.13 and `clang` at least 3.9. You should set the `LIBCLANG_PATH` environment
+variable to the directory containing `libclang.so.1`.
+
+For example, in Ubuntu 16.04:
+```
+sudo apt-get install autoconf2.13 clang-4.0
+LIBCLANG_PATH=/usr/lib/llvm-4.0/lib/ cargo +nightly build
+```
+
+The `LIBCLANG_PATH` environment variable is only needed by codegen when compiling `mozjs`.
+Once it is built, you can use the usual cargo commands such as `cargo +nightly test`.

